connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedKafkaCluster.java

```
@@ -0,0 +1,321 @@
+
+    @Override
+    protected void after() {
+        stop();
+    }
+
+    private void start() throws IOException {
+        zookeeper = new EmbeddedZookeeper();
+
+        brokerConfig.put(KafkaConfig$.MODULE$.ZkConnectProp(), zKConnectString());
+        brokerConfig.put(KafkaConfig$.MODULE$.PortProp(), 0); // pick a random port
+
+        putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.HostNameProp(), "localhost");
+        putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.DeleteTopicEnableProp(), true);
+        putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.GroupInitialRebalanceDelayMsProp(), 0);
+        putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.OffsetsTopicReplicationFactorProp(), (short) 1);
+        putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.AutoCreateTopicsEnableProp(), false);
+
+        Object listenerConfig = brokerConfig.get(KafkaConfig$.MODULE$.InterBrokerListenerNameProp());
+        if (listenerConfig != null) {
+            listenerName = new ListenerName(listenerConfig.toString());
+        }
+
+        for (int i = 0; i < brokers.length; i++) {
+            brokerConfig.put(KafkaConfig$.MODULE$.BrokerIdProp(), i);
+            brokerConfig.put(KafkaConfig$.MODULE$.LogDirProp(), createLogDir());
+            brokers[i] = TestUtils.createServer(new KafkaConfig(brokerConfig, true), time);
+        }
+
+        Map<String, Object> producerProps = new HashMap<>();
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers());
+        producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer");
+        producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer");
+        producer = new KafkaProducer<>(producerProps);
+    }
+
+    private void stop() {
+
+        try {
+            producer.close();
+        } catch (Exception e) {
+            log.error("Could not shutdown producer ", e);
```

same as mentioned elsewhere, none of these errors should just be swallowed and logged since they won't be noticed, they should cause the test to fail (or have the test marked with expected failures if they are expected to fail)